odoo/models.py

```diff
@@ -3633,6 +3633,26 @@ def get_base_url(self):
             raise ValueError("Expected singleton or no record: %s" % self)
         return self.env['ir.config_parameter'].sudo().get_param('web.base.url')
 
+    def _parse_companies(self, companies):
+        if companies is None:
+            return self.env.company.ids
+        elif isinstance(companies, BaseModel):
+            return companies.ids
+        elif isinstance(companies, (list, tuple)):
+            return companies
+        return [companies]
+
+    def _check_company_domain_parent_of(self, companies=None):
```

Comment:
Could you move this method outside of the class?
That's a component which is reused in many classes, but does not belong to `BaseModel`...

Same remark for `_parse_companies`: it's a helper, not a method to be overridden.
By the way, its name is confusing.  Its name should reflect _what_ it does, not _how_ it does it.
It converts `companies` into a list of company ids, and does not even "parse" anything.
I suggest `to_company_ids(...)` instead.